### PR TITLE
FileInfoStore migration

### PR DESCRIPTION
**Pull request by rvillablanca**
#### Summary

Migration of `FileInfoStore` to return plain errors.

Fixes https://github.com/mattermost/mattermost-server/issues/15036

### CODE REVIEW

File: store/sqlstore/file_info_store.go [hunk 1]
```diff
@@ -293,28 +286,28 @@ func (fs SqlFileInfoStore) PermanentDeleteBatch(endTime int64, limit int64) (int
...
 	sqlResult, err := fs.GetMaster().Exec(query, map[string]interface{}{"CreatorId": userId})
 	if err != nil {
-		return 0, model.NewAppError("SqlFileInfoStore.PermanentDeleteByUser", "store.sql_file_info.PermanentDeleteByUser.app_error", nil, ""+err.Error(), http.StatusInternalServerError)
+		return 0, errors.Wrapf(err, "failed to delete FileInfo with creatorId=%s", userId)
 	}
 
 	rowsAffected, err := sqlResult.RowsAffected()
 	if err != nil {
-		return 0, model.NewAppError("SqlFileInfoStore.PermanentDeleteByUser", "store.sql_file_info.PermanentDeleteByUser.app_error", nil, ""+err.Error(), http.StatusInternalServerError)
+		return 0, errors.Wrapf(err, "unable to retrive rows affected")
```
Comment (agnivade) [inline store/sqlstore/file_info_store.go:310]: typo at retrieve

File: store/sqlstore/file_info_store.go [hunk 2]
```diff
@@ -76,17 +76,16 @@ func (fs SqlFileInfoStore) Get(id string) (*model.FileInfo, *model.AppError) {
...
 	}
 	return info, nil
 }
 
-func (fs SqlFileInfoStore) GetWithOptions(page, perPage int, opt *model.GetFileInfosOptions) ([]*model.FileInfo, *model.AppError) {
+func (fs SqlFileInfoStore) GetWithOptions(page, perPage int, opt *model.GetFileInfosOptions) ([]*model.FileInfo, error) {
 	if perPage < 0 || page < 0 {
-		return nil, model.NewAppError("SqlFileInfoStore.GetWithOptions",
-			"store.sql_file_info.get_with_options.app_error", nil, fmt.Sprintf("page=%d and perPage=%d must be non-negative", page, perPage), http.StatusBadRequest)
+		return nil, store.NewErrInvalidInput("FileInfo", "<perPage, page>", fmt.Sprintf("<%d, %d>", perPage, page))
```
Comment (agnivade) [inline store/sqlstore/file_info_store.go:88]: Wondering if "ErrLimitExceeded" is a better choice here?

File: store/sqlstore/file_info_store.go [hunk 3]
```diff
@@ -293,28 +286,28 @@ func (fs SqlFileInfoStore) PermanentDeleteBatch(endTime int64, limit int64) (int
...
 	sqlResult, err := fs.GetMaster().Exec(query, map[string]interface{}{"EndTime": endTime, "Limit": limit})
 	if err != nil {
-		return 0, model.NewAppError("SqlFileInfoStore.PermanentDeleteBatch", "store.sql_file_info.permanent_delete_batch.app_error", nil, ""+err.Error(), http.StatusInternalServerError)
+		return 0, errors.Wrap(err, "failed to delete FileInfos in batch")
 	}
 
 	rowsAffected, err := sqlResult.RowsAffected()
 	if err != nil {
-		return 0, model.NewAppError("SqlFileInfoStore.PermanentDeleteBatch", "store.sql_file_info.permanent_delete_batch.app_error", nil, ""+err.Error(), http.StatusInternalServerError)
+		return 0, errors.Wrapf(err, "unable to retrive rows affected")
```
Comment (agnivade) [inline store/sqlstore/file_info_store.go:294]: typo

File: store/sqlstore/file_info_store.go (hunk 2 above)
Comment (rvillablanca) [reply store/sqlstore/file_info_store.go:88]: sure!

File: app/file.go [hunk 4]
```diff
@@ -1104,11 +1118,33 @@ func (a *App) generatePreviewImage(img image.Image, previewPath string, width in
...
+	}
+
+	return fileInfo, nil
 }
 
 func (a *App) GetFileInfos(page, perPage int, opt *model.GetFileInfosOptions) ([]*model.FileInfo, *model.AppError) {
-	return a.Srv().Store.FileInfo().GetWithOptions(page, perPage, opt)
+	fileInfos, err := a.Srv().Store.FileInfo().GetWithOptions(page, perPage, opt)
+	if err != nil {
+		var ltErr *store.ErrLimitExceeded
```
Comment (jespino) [inline app/file.go:1138]: Should we handling here the `NewErrInvalidInput` error and return a bad request instead of an internal server error?

Comment (rvillablanca) [reply app/file.go:1138]: yep Jesus, we just changed to `ErrLimitExceeded` but in the app layer we are returning the bad request status, so it is ok

File: store/sqlstore/file_info_store.go [hunk 5]
```diff
@@ -131,8 +132,7 @@ func (fs SqlFileInfoStore) GetWithOptions(page, perPage int, opt *model.GetFileI
 	case model.FILEINFO_SORT_BY_SIZE:
 		query = query.OrderBy("FileInfo.Size " + sortDirection)
 	default:
-		return nil, model.NewAppError("SqlFileInfoStore.GetWithOptions",
-			"store.sql_file_info.get_with_options.app_error", nil, "invalid sort option", http.StatusBadRequest)
+		return nil, store.NewErrInvalidInput("FileInfo", "<sortOption>", opt.SortBy)
```
Comment (jespino) [inline store/sqlstore/file_info_store.go:135]: What about this error case?

Comment (rvillablanca) [reply store/sqlstore/file_info_store.go:135]: you are right, thanks!

**Comment by rvillablanca** [comment]
Now it should be ok @jespino good catch!

**Comment by rvillablanca** [comment]
@jasonblais we have conflicts but I will resolve them after the QA review. It doesn't make sense to revolve them on each `master` update. The same applies to others PRs :)

**Comment by jasonblais** [comment]
@ogi-m friendly reminder to help with QA review, thanks!

**Comment by agnivade** [comment]
Needs EE changes.

**Pull request closed without merging**